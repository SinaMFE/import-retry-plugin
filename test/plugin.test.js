import webpack from './helpers/compiler';
describe('Loader', () => {
  test('Defaults', async () => {
    const stats = await webpack('fixture.js',{output:"/dist/"});
    expect(stats.toString()).toMatchSnapshot();
  });
});
