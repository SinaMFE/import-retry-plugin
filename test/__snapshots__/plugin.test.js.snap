// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Defaults 1`] = `"!function(e){function r(r){for(var n,o,i=r[0],u=r[1],a=0,l=[];a<i.length;a++)o=i[a],t[o]&&l.push(t[o][0]),t[o]=0;for(n in u)Object.prototype.hasOwnProperty.call(u,n)&&(e[n]=u[n]);for(c&&c(r);l.length;)l.shift()()}var n={},t={0:0};function o(r){if(n[r])return n[r].exports;var t=n[r]={i:r,l:!1,exports:{}};return e[r].call(t.exports,t,t.exports,o),t.l=!0,t.exports}o.e=function e(r){var n=[],i=t[r];if(0!==i)if(i)n.push(i[2]);else{var u=new Promise(function(e,n){i=t[r]=[e,n]});n.push(i[2]=u);var a,c=document.createElement(\\"script\\");c.charset=\\"utf-8\\",c.timeout=120,o.nc&&c.setAttribute(\\"nonce\\",o.nc),window.retryerror||(window.retryerror={});var l=function(e){return o.p+\\"\\"+({}[e]||e)+\\".bundle.js\\"}(r);1==window.retryerror[r]&&(l=l.indexOf(\\"?\\")>0?l+\\"&t=\\"+(new Date).getTime():l+\\"?t=\\"+(new Date).getTime()),c.src=l,a=function(n){c.onerror=c.onload=null,clearTimeout(f);var o=t[r];if(0!==o){if(o){var i=n&&n.target&&n.target.src;if(1==window.retryerror[r]){window.onChunkError&&window.onChunkError(i,2);var u=n&&(\\"load\\"===n.type?\\"missing\\":n.type),a=new Error(\\"Loading chunk \\"+r+\\" failed.\\\\n(\\"+u+\\": \\"+i+\\")\\");return a.type=u,a.request=i,o[1](a),void(t[r]=void 0)}}window.retryerror[r]=1,window.onChunkError&&window.onChunkError(i,1),t[r]=void 0,e(r)}};var f=setTimeout(function(){a({type:\\"timeout\\",target:c})},12e4);c.onerror=c.onload=a,document.head.appendChild(c)}return Promise.all(n)},o.m=e,o.c=n,o.d=function(e,r,n){o.o(e,r)||Object.defineProperty(e,r,{enumerable:!0,get:n})},o.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},o.t=function(e,r){if(1&r&&(e=o(e)),8&r)return e;if(4&r&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(o.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:e}),2&r&&\\"string\\"!=typeof e)for(var t in e)o.d(n,t,function(r){return e[r]}.bind(null,t));return n},o.n=function(e){var r=e&&e.__esModule?function(){return e.default}:function(){return e};return o.d(r,\\"a\\",r),r},o.o=function(e,r){return Object.prototype.hasOwnProperty.call(e,r)},o.p=\\"\\",o.oe=function(e){throw console.error(e),e};var i=window.webpackJsonp=window.webpackJsonp||[],u=i.push.bind(i);i.push=r,i=i.slice();for(var a=0;a<i.length;a++)r(i[a]);var c=u;o(o.s=0)}([function(e,r,n){n.e(1).then(n.bind(null,1)).then(function(e){console.log(e)})}]);"`;
